mining/src/main/java/net/runelite/client/plugins/mining/MiningPlugin.java

```diff
@@ -315,4 +374,21 @@ private boolean inMiningGuild()
 	{
 		return client.getLocalPlayer().getWorldLocation().getRegionID() == MINING_GUILD_REGION;
 	}
+
+	private boolean isAtDaeyaltEssence()
+	{
+		return client.getLocalPlayer().getWorldLocation().getRegionID() == DAEYALT_ESSENCE_REGION;
+	}
+
+	private void locateActiveDaeyaltEssence()
+	{
+		for (GameObject gameObject : gameObjectQuery.result(client))
```

Comment:
so, to reiterate from above, this entire method could be replaced with:
`private void locateActiveDaeyalt()
{
    activeDaeyalt = new GameObjectQuery().idEquals(ACTIVE_DAEYALT_ESSENCE).result(client).first();
}`